plugin/src/test/java/io/jenkins/plugins/casc/misc/ExportImportRoundTripAbstractTest.java

```diff
@@ -0,0 +1,258 @@
+package io.jenkins.plugins.casc.misc;
+
+import com.gargoylesoftware.htmlunit.WebRequest;
+import com.gargoylesoftware.htmlunit.WebResponse;
+import com.gargoylesoftware.htmlunit.util.NameValuePair;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import io.jenkins.plugins.casc.ConfiguratorException;
+import io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.logging.Level;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.LoggerRule;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+import static com.gargoylesoftware.htmlunit.HttpMethod.POST;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ *
+ */
+public abstract class ExportImportRoundTripAbstractTest {
```

Comment:
Could you explain the reasoning for a restartable Jenkins with roundtrip configuration? 
What's the issue? What are you trying to solve?
Why can't you use the existing test method that others have done?

Reply:
I'm trying to test easily for a plugin author:
1. The instance is configured with his/her piece of yaml
2. The full configuration exported can be applied (not currently - added a hack)
3. The instance is configured using the default config file.

Reply:
I think the idea is great. I would change maybe some points of the implementation since I've said in the general comment, but that should help to any maintainer to test the compatibility of their plugins.